packages/application-shell/src/components/application-page-title/application-page-title.tsx

```diff
@@ -0,0 +1,131 @@
+  suffix: string;
+  paths: string[];
+  location: TEnhancedLocation<Query>;
+};
+type Props<Query extends {} = {}> = {
+  // Allow to render a custom page title.
+  /**
+   * Overrides default page title with the format:
+   * <?resource_name(manually inputed)> - <?page_location> - <application_identifier> - <project_key> - Merchant Center
+   */
+  renderPageTitle?: (breadcrumb: Breadcrumb<Query>) => string;
+  children: React.ReactNode;
+};
+
+const maxTitleCharLength = 24;
+
+const compilePath = (paths: string[], hasTruncatedPaths = false): string[] => {
+  if (hasTruncatedPaths) {
+    return [...paths.slice(0, 1), '...', ...paths.slice(1)];
+  }
+  return paths;
+};
+
+const getLimitedPaths = (
+  paths: string[],
+  hasTruncatedPaths = false
+): string[] => {
+  // We want to keep a minimum of 2 paths
+  if (paths.length <= 2) {
+    return compilePath(paths, hasTruncatedPaths);
+  }
+
+  // Calculate the length projection
+  const pathsProjection = paths.join(' - ');
+
+  // If the length exceeds the max allowed length, we need to remove one of the paths
+  // from the list, except from the first and last.
+  if (pathsProjection.length > maxTitleCharLength) {
+    return getLimitedPaths(
+      [
+        // we always remove the second item
+        ...paths.slice(0, 1),
+        ...paths.slice(2),
+      ],
+      true
+    );
+  }
+  return compilePath(paths, hasTruncatedPaths);
+};
+
+const isStaticPath = (path: string) => {
+  const reservedStaticPaths = ['account', 'login'];
+  return reservedStaticPaths.includes(path);
+};
+
+/**
+ * Converts a URI path into a default mapping.
+ * Format is something like this: <?page_location> - <project_key> - Merchant Center
+ *
+ * Example:
+ *   /almond-40/products/<id> should display as <product name> - Products - almond-40 - Merchant Center
+ */
+
+const compilePageTitle = <Query extends {}>(
+  location: TEnhancedLocation<Query>
+): Breadcrumb<Query> => {
+  const [, projectKeyOrStaticPath, ...locationPaths] =
+    location.pathname.split('/');
+
+  const suffix = isStaticPath(projectKeyOrStaticPath)
+    ? ' Merchant Center'
+    : `${projectKeyOrStaticPath} - Merchant Center`;
+
+  const paths = isStaticPath(projectKeyOrStaticPath)
+    ? getLimitedPaths([projectKeyOrStaticPath])
+    : getLimitedPaths(
+        locationPaths.length === 0 ? [projectKeyOrStaticPath] : locationPaths
+      );
+  return {
+    suffix,
+    paths,
+    location,
+  };
+};
+
+const convertToReadablePrefix = (prefix: string): string => {
+  switch (prefix) {
+    case 'app-kit-playground':
+      return 'App Kit Playground';
+    case 'dashboard':
+      return 'DashBoard';
+    case 'products':
+      return 'Products';
+    case 'orders':
+      return 'Orders';
+    case 'settings':
+      return 'Settings';
+    case 'projects':
+      return 'Projects';
+    case 'login':
+      return 'Login';
+    case 'account':
+      return 'Account';
+    default:
+      return startCase(prefix);
```

Comment:
👋🏼 Thanks for working on this. I am frankly new to the topic but am wondering if this is something we should allow configuration of via a Custom Application's config. Then we might not need this map here.

Reply:
Oh, thanks for pointing this out. I will look into it.

Reply:
No need to configure this, this mapping is not really necessary.

I would suggest to focus on this comment first: https://github.com/commercetools/merchant-center-application-kit/pull/2758#issuecomment-1216602555

Then things will fall nicely into place, following that path.